src/store/modules/product/actions.ts

```diff
@@ -40,6 +40,30 @@ const actions: ActionTree<ProductState, RootState> = {
     return resp;
   },
 
+  async fetchProduct({ commit, state }, { productId }) {
+    const cachedProduct = Object.keys(state.cached);
+    if(cachedProduct.includes(productId)) {
+      return productId;
+    }
+    else {
+      let resp;
+      try {
+        resp = await ProductService.fetchProducts({
+          "filters": [`productId: ( ${productId} )`]
+        })
+        if(resp.status == 200 && resp.data.response?.numFound > 0 && !hasError(resp)) {
+          const product = resp.data.response.docs[0];
+          commit(types.PRODUCT_ADD_TO_CACHED, { product });
```

Comment:
```suggestion
          commit(types.PRODUCT_ADD_TO_CACHED, product);
```
Update the mutation accordingly.

Reply:
Improved